test/unit/APIConsumer_unit_test.js

```diff
@@ -27,6 +28,14 @@ skip.if(!developmentChains.includes(network.name)).
       if (await autoFundCheck(apiConsumer.address, networkName, linkTokenAddress, additionalMessage)) {
         await hre.run("fund-link", { contract: apiConsumer.address, linkaddress: linkTokenAddress })
       }
+
+      const MockOracle = await deployments.get("MockOracle");
+      mockOracle = await ethers.getContractAt("MockOracle", MockOracle.address);
+    })
+
+    afterEach(async () => {
+      mockOracle.removeAllListeners()
+      apiConsumer.removeAllListeners()
     })
```

Comment:
I don't think there is a _listener_ for `apiConsumer` contract object, correct me if I am wrong

```suggestion
    afterEach(async () => {
      mockOracle.removeAllListeners()
    })
 ```